### PR TITLE
#21669 support schemas + some tables from schemas dump process

### LINKED ISSUE #21669: Backup of several schemes with selective set of tables is wrong
### Description

I try to backup 6 schemes from one database, but at one of them I need only several tables, not all. So after Selection of needed schemas and tables I start backup and see that DBeaver save every table, whatever I selected.

### DBeaver Version

Community edition 23.2.3

### Operating System

Windows 10 Pro x64

### Database and driver

1. Database PostgreSQL 14.4 (Ubuntu 14.4-1.pgdg20.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit
2. Driver (PostgreSQL JDBC Driver) org.postgresql:postgresql:RELEASE 42.5.2

### Steps to reproduce

1. Connecto to PostgreSQL server
2. Open right click mouse menu -> Instruments -> Backup (may be not exactly because I translated from Russian)
3. Select schemas
4. Select Needed tables from last scheme (about 3 from ~40 tables)
5. Continue -> select file path and name
6. Start

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LonwoLonwo**
- Now, in case we have checked schemas + a checked schema without tables (like in the picture):
one file for all entire schemas will be created
and one file for every schema with partially checked tables/views will be created
- special note added to the wizard page

2023-10-30 13_49_58-PostgreSQL dump

2023-11-15 18_19_39-DBeaver Ultimate 23 2 5 - my_schema

So, please check:

- [x] case from the ticket
- [x] one table from one schema dump
- [x] one entire schema dump
- [x] a few whole schemas dump
- [x] a few whole schemas + one schema partially dump
- [x] performance test on big (many tables) schemas

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/AbstractNativeToolHandler.java [hunk 1]
```diff
@@ -175,9 +175,18 @@ protected boolean validateTaskParameters(DBTTask task, SETTINGS settings, Log lo
         return true;
     }
 
-    abstract protected java.util.List<String> getCommandLine(SETTINGS settings, PROCESS_ARG arg) throws IOException;
+    abstract protected java.util.List<String> getCommandLine(
+        SETTINGS settings,
+        PROCESS_ARG arg,
+        @NotNull DBRProgressMonitor monitor
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/AbstractNativeToolHandler.java]: Just a nitpick: I'd pass the monitor as the very first argument.

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/AbstractNativeToolHandler.java]: Usually, I do this way, yes. But here are generics. I really think that they need to be first.

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/tasks/PostgreDatabaseBackupHandler.java [hunk 2]
```diff
@@ -128,26 +134,56 @@ public void fillProcessParameters(PostgreDatabaseBackupSettings settings, Postgr
...
+                        List<PostgreTableBase> tablesAndViews = schema.getTableCache().getAllObjects(monitor, schema);
+                        for (PostgreTableBase object : tablesAndViews) {
+                            if (!values.contains(object) && (object instanceof PostgreTableRegular || object instanceof PostgreView)) {
+                                excludeTables.add(object);
+                            }
+                        }
+                    } catch (DBException e) {
+                        log.debug("Can't read tables from schema " + schema.getName());
+                    }
+                }
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/tasks/PostgreDatabaseBackupHandler.java]: Can we refactor it to a separate method?

**Comment by LonwoLonwo** [comment]
We can probably investigate the ability to do a command a few times to create a few outputs and then make one file from it.

**Comment by LonwoLonwo** [comment]
Logic changed

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/tools/PostgreBackupWizardPageObjects.java [hunk 3]
```diff
@@ -308,24 +308,28 @@ public void saveState() {
...
-                if (!schemas.contains(schema)) {
-                    schemas.add(schema);
-                }
-                if (checkedTables != null) {
-                    // Only a few tables checked
-                    tables.addAll(checkedTables);
+                if (checkedTables == null) {
+                    // All tables checked, we can add this schema in the full dump
+                    entireSchemas.add(schema);
+                    continue;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/tools/PostgreBackupWizardPageObjects.java:319]: It's a personal taste thingy, but I'd prefer to use `else` instead to ease understanding code flow.

**Review by ShadelessFox — APPROVED** [review]

**Comment by DariaMarkaryan** [comment]
Verified

**Pull request merged**